ui/app/src/nh-config/dimension-list.ts

```diff
@@ -0,0 +1,265 @@
+    }
+  }
+
+  renderRangeDetails(range: Range & { range_eh: EntryHash } | undefined) : TemplateResult {
+    if(typeof range == "undefined") return html``;
+    return html`
+      <nh-card .theme=${"light"} .title=${range.name} >
+        <h3><em>Kind:</em></h3>
+        <div>${Object.keys(range.kind)[0]}</div>
+        <h3><em>Min:</em></h3>
+        <span>${Object.values(range.kind)[0].min}</span>
+        <h3><em>Max:</em></h3>
+        <span>${Object.values(range.kind)[0].max}</span>
+      </nh-card>
+    `
+  }
+
+  resetSelectedInputDimensionIndex() {
+    this._selectedInputDimensionIndex = 0;
+  }
+
+  render() {
+    return html`
+      <nh-card .theme=${"light"} .title=${"Existing Input Dimensions"} .textSize=${"sm"}>
+        <div class="content">
+          ${
+            typeof this._dimensionEntries == 'undefined' || this._dimensionEntries.length == 0
+              ? "No dimensions available"
+              : html`<div style="display:flex; flex-direction: column; gap: 8px;">
+                  ${this._dimensionEntries.filter((dimension: Dimension) => !dimension.computed)
+                    .reverse()
+                    .map((dimension: Dimension & { dimension_eh: EntryHash} , dimensionIndex: number) => {
+                      return keyed(encodeHashToBase64(dimension.dimension_eh), html`
+                          <nh-card 
+                            class="nested-card ${classMap({
+                                selected: this._selectedInputDimensionIndex == dimensionIndex,
+                              })}" .theme=${"dark"}
+                            .heading=${dimension.name}
+                            .textSize=${"sm"}
+                          >
+                            <h1>Range: </h1>
+                            ${this._rangeEntries?.length && this.renderRangeDetails(this._rangeEntries.find((range: Range & { range_eh: EntryHash }) => encodeHashToBase64(range.range_eh) === encodeHashToBase64(dimension.range_eh)))}
+                            
+                            ${this.filteredMethodInputDimensions.length > 0 && this.filteredMethodInputDimensions.some(({name}) => name === dimension.name)
+                              ? html`<h2>Methods using this dimension: </h2>
+                                ${this.filteredMethodInputDimensions.map(({methodEh, name}) => {
+                                  return name == dimension.name
+                                    ? html`${generateHashHTML(methodEh as string)}`
+                                    : null
+                                })}
+                              `
+                              :html`<h1>No methods for this dimension</h1>
+
+                              ${this._selectedInputDimensionIndex !== dimensionIndex
+                                ? html`<nh-button .size=${"sm"} .variant=${"warning"} @click=${() => {
```

Comment:
Make this a named arrow function in the class. There should not be any long in-line closures in an html literal.